Doc/library/ast.rst

```diff
@@ -378,5 +378,24 @@ to stdout.  Otherwise, the content is read from stdin.
 
 .. seealso::
 
-    `Green Tree Snakes <https://greentreesnakes.readthedocs.io/>`_, an external documentation resource, has good
-    details on working with Python ASTs.
+    `Green Tree Snakes <https://greentreesnakes.readthedocs.io/>`_, an external
+    documentation resource, has good details on working with Python ASTs.
+
+    `ASTTokens <https://asttokens.readthedocs.io/en/latest/user-guide.html>`_
+    annotates Python ASTs with the positions of tokens and text in the source
+    code that generated them. This is helpful for tools that make source code
+    transformations.
+
+    `leoAst.py <http://leoeditor.com/appendices.html#leoast-py>`_ unifies the
+    token-based and parse-tree-based views of python programs by inserting
+    two-way links between tokens and ast nodes.
```

Comment:
Is this a commonly-known tool? I never heard of it, until today.

Reply:
@edreamleo This is a reasonable question. What shall we say?

Reply:
@isidentical This is a perfectly reasonable question and it would be reasonable to omit the link to leoAst.py on the grounds of "obscurity". I'll make a pitch for inclusion below.

The first widely public announcement appeared [here](https://mail.python.org/archives/list/python-announce-list@python.org/thread/K2XM2O57QBRWQOYE6UDTTSQJROO2BDBC/) on July 25. I discussed it extensively on [leo-editor](https://groups.google.com/forum/#!forum/leo-editor) late last year and earlier this year, but surely not many people read about it. [This comment](https://github.com/leo-editor/leo-editor/issues/1440#issuecomment-574145510) gives an overview of the history of the project.

**Why many people may be interested in leoAst.py**

In a private communication, Guido said, "I think your approach is fascinating -- visit the AST in token order and match up with the corresponding list of tokens, dealing with the various exceptional cases along the way."

In [this comment](https://github.com/leo-editor/leo-editor/issues/1565#issuecomment-654904747), the section called "Applicability and Importance" contains:

Many python developers will find [asttokens](https://github.com/gristlabs/asttokens) meets all their needs. asttokens is well documented and easy to use. Nevertheless, two-way links are significant additions to python's [tokenize](https://docs.python.org/3/library/tokenize.html) and [ast](https://docs.python.org/3/library/ast.html) modules:

- Links from tokens to nodes are assigned to the nearest possible ast node, not the nearest statement, as in asttokens. Links can easily be reassigned, if desired.

- The TOG and TOT classes are intended to be the foundation of tools such as [**fstringify**](https://pypi.org/project/fstringify/) and [**black**](https://pypi.org/project/black/).

- The TOG class solves real problems, such as this [FAQ](https://stackoverflow.com/questions/16748029/).

Reply:
Oh, that looks like a pretty good approach but as far as I can see, it looks like a small utility file in a very big project. I guess that you are the maintainer of that project, so can you make any assertions about whether that simple utility will go into the official API of your editor and will be maintained or not?

Reply:
Yes, leoAst.py is part of Leo, a much bigger project. Leo's beautify and fstringify commands, defined in leoBeautify.py, use leoAst.py. All of Leo's sources have been processed with those commands. 

I'll maintain leoAst.py while I am able. However, I am about to turn 71 years old. [This post](https://groups.google.com/forum/#!topic/leo-editor/0YTNRcphKo4) expresses my confidence that Leo will outlive me. A few days ago I wrote [this post](https://groups.google.com/forum/#!topic/leo-editor/LFUGr9TzZKM), asking for a volunteer to maintain leoAst.py. It's not a huge job, as described [here](http://leoeditor.com/appendices.html#maintaining-leoast-py).

Reply:
Thank you for the explanations, and all the work you have freely put into this. I'd be glad to see leoAst.py as a resource over there!